dask/dot.py

```diff
@@ -158,7 +159,7 @@ def to_graphviz(
             func_name = name((k, "function")) if not collapse_outputs else k_name
             if collapse_outputs or func_name not in seen:
                 seen.add(func_name)
-                attrs = function_attributes.get(k, {})
+                attrs = copy.copy(function_attributes.get(k, {}))
```

Comment:
It'd be better to do a copy at the very start, around https://github.com/dask/dask/pull/5996/files#diff-978782aa4edf590e4947a357f6d351b3R143.

Add an `else: function_attributes = copy.copy(function_attributes, deep=True)`.

Seems like data_attributes has a similar issue, if you could fix that up while your in there.

Reply:
The problem about the test is that I don't know how to create an argument for which the problem happens. If I just assign a new dictionary the problem does not happen as you can see [here](https://github.com/dask/dask/issues/5995#issue-577906926).

Reply:
I think this roughly reproduces the original issue

```python
In [1]: import operator

In [2]: import dask.dot

In [3]: def inc(x): return x + 1

In [4]: dsk = {"a": (inc, 1), "b": (inc, 2), "c": (operator.add, 'a', 'b')}

In [5]: attrs = {"a": {}}

In [6]: dask.dot.to_graphviz(dsk, function_attributes=attrs)
Out[6]: <graphviz.dot.Digraph at 0x10dc80090>

In [7]: attrs
Out[7]: {'a': {'label': 'a', 'shape': 'circle'}}
```

Can you try writing a test around that?

Reply:
Okay, I wrote a test for that overwriting behavior, BUT this doesn't change the wrong label behavior I talked about in #5995. 

There still is bug here:
https://github.com/dask/dask/blob/7b4f90f0a708cbb97ebb498c78f7931d82ec99db/dask/dot.py#L163-L165

When `attrs` are being set, this also changes `function_attributes`. For consecutive iterations `attrs`, will always be what they were set to on the first iteration, because `setdefault` is being used. Do you understand what I mean?

Can I change it back to https://github.com/dask/dask/pull/5996/commits/1b7eb8e35350e0618eb914ad572808630a3694a6?

Reply:
Not sure, I'm not really familiar with this section.

But initially, I would say that the label issue and mutating input dict issue can be solved separately.